ada/extensions/src/libadalang-helpers.ads

```diff
@@ -205,6 +205,11 @@ package Libadalang.Helpers is
             Default_Val => Null_Unbounded_String,
             Help        => "Project file to use");
 
+         package Process_Full_Project_Tree is new Parse_Flag
+           (Parser, "-U", "--recursive",
```

Comment:
In other GPR tools, `-U` means to disable the recursive behavior: please remove this short option.

Reply:
@setton ping

Reply:
Here is what [gprbuild](
https://docs.adacore.com/gprbuild-docs/html/gprbuild_ug/building_with_gprbuild.html#switches) does:

`-U (Compile all sources of all projects)` 

If there are sources specified on the command line, only compile these sources. If there are no sources specified on the command line, compile all the sources of all the projects in the project tree.

Reply:
As discussed live, I think it’s still misleading when comparing to gprbuild:

* by default, gprbuild processes only closures of mains, which is not what this framework does (it doest not even compute these closures)
* by default, gprbuild processes the root project and all its non-externally built dependencies, whereas this framework processes only the root project by default
* hence while `-U` in gprbuild means “stop considering closures, look at either the files given on the command line, or at the whole project closure otherwise”, here it means “look at subprojects recursively”

Because of this, I think we should just omit a short option, here.

Reply:
I'll remove the short option because it's not the battle for this particular PR. However, I disagree with your analysis - the decision has been made to standardise on `-U` for all project-aware tools for exactly what it's used here for - whether or not they are by default considering only the closure of mains.

Reply:
Ok. If it’s an already fought battle, don’t bother. Thank you for the context.